src/teams.rs

```diff
@@ -101,25 +102,78 @@ impl Data {
+        let mut raw_subteams = Vec::new();
         let mut wgs = Vec::new();
         let mut project_groups = Vec::new();
+
+        let superteams: HashMap<_, _> = self
+            .teams
+            .iter()
+            .filter(|team| matches!(team.kind, TeamKind::Team))
+            .filter_map(|team| Some((team.name.clone(), team.subteam_of.clone()?)))
+            .collect();
+
         self.teams
             .into_iter()
             .filter(|team| team.website_data.is_some())
-            .filter(|team| team.subteam_of.as_ref() == Some(&main_team.name))
+            .filter(|team| {
+                // For teams find not only direct subteams but also transitive ones.
+                if let TeamKind::Team = team.kind {
+                    let mut team = &team.name;
+
+                    // The graph of teams is guaranteed to be acyclic by the the CI script of
+                    // the team repository. Therefore this loop has to terminate eventually.
+                    while let Some(superteam) = superteams.get(team) {
+                        if superteam == &main_team.name {
+                            return true;
+                        }
+
+                        team = superteam;
+                    }
+
+                    return false;
+                }
+
+                team.subteam_of.as_ref() == Some(&main_team.name)
+            })
             .for_each(|team| match team.kind {
-                TeamKind::Team => subteams.push(team),
+                TeamKind::Team => raw_subteams.push(team),
                 TeamKind::WorkingGroup => wgs.push(team),
                 TeamKind::ProjectGroup => project_groups.push(team),
                 _ => {}
             });
 
-        subteams.sort_by_key(|team| Reverse(team.website_data.as_ref().unwrap().weight));
+        raw_subteams.sort_by_key(|team| Reverse(team.website_data.as_ref().unwrap().weight));
         wgs.sort_by_key(|team| Reverse(team.website_data.as_ref().unwrap().weight));
         project_groups.sort_by_key(|team| Reverse(team.website_data.as_ref().unwrap().weight));
 
+        // Lay out subteams according to their hierarchy.
+        // Superteams come first and subteams come after, recursively.
+        // The ordering of sibling teams is unaffected, they remain sorted first by weight, then
+        // alphabetically if we assume that the Team API serves the teams alphabetically.
+        let mut subteams = Vec::with_capacity(raw_subteams.len());
+        lay_out_subteams_hierarchically(&mut subteams, None, &main_team.name, &raw_subteams);
+
+        fn lay_out_subteams_hierarchically<'a>(
+            result: &mut Vec<Team>,
+            team: Option<&Team>,
+            main_team: &str,
+            subteams: &[Team],
+        ) {
+            let name = team.map(|team| team.name.as_str()).or(Some(main_team));
+
+            if let Some(team) = team {
+                result.push(team.clone());
+            }
+
+            for subteam in subteams {
```

Comment:
Not super happy about this loop inside this recursive function. Complexity could be better but input size should be relatively low. Let me know if there's a nice way to improve the algorithmic complexity.